uint16 enc_cnt_l
uint16 enc_cnt_r